demo-code/factorial.js

```
@@ -0,0 +1,14 @@
+// This function calculates the factorial of a number
+function factorial(n) {
+	if (n < 0) {
+		return "Error: Negative input is not allowed";
```

Instead of returning string, we can throw error?